src/Lib.hs

```diff
@@ -456,23 +460,31 @@ splitWithSeparator :: Char -> String -> [String]
 splitWithSeparator a as = map T.unpack (T.split (== a) $ T.pack as)
 
 getMergedGitRepoAvailabilitiesFromFrameworkAvailabilities :: InvertedRepositoryMap -> [FrameworkAvailability] -> [GitRepoAvailability]
-getMergedGitRepoAvailabilitiesFromFrameworkAvailabilities reverseRomeMap = (>>=mergeRepoAvailabilities) . groupAvailabilities . map (getGitRepoAvailabilityFromFrameworkAvailability)
+getMergedGitRepoAvailabilitiesFromFrameworkAvailabilities reverseRomeMap = concatMap mergeRepoAvailabilities . groupAvailabilities . getGitRepoAvalabilities
   where
-    repoNameForFrameworkName frameworkName = maybe (GitRepoName . unFrameworkName $ frameworkName) id (M.lookup frameworkName reverseRomeMap)
+    getGitRepoAvalabilities :: [FrameworkAvailability] -> [GitRepoAvailability]
+    getGitRepoAvalabilities = fmap getGitRepoAvailabilityFromFrameworkAvailability
+
+    repoNameForFrameworkName :: FrameworkName -> GitRepoName
+    repoNameForFrameworkName frameworkName = fromMaybe (GitRepoName . unFrameworkName $ frameworkName) (M.lookup frameworkName reverseRomeMap)
```

Comment:
@r-peck are you sure you want to create a "default" GitRepoName rather then returning a Maybe?